SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,529 @@
+  inlet_species = config->GetInlet_SpeciesVal(Marker_Tag);
+
+  /*--- Loop over all the vertices on this boundary marker ---*/
+
+  SU2_OMP_FOR_STAT(OMP_MIN_SIZE)
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+
+    if (geometry->nodes->GetDomain(iPoint)) {
+      nodes->SetSolution_Old(iPoint, inlet_species);
+
+      LinSysRes.SetBlock_Zero(iPoint);
+
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        nodes->SetVal_ResTruncError_Zero(iPoint, iVar);
+      }
+
+      /*--- Includes 1 in the diagonal ---*/
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        auto total_index = iPoint * nVar + iVar;
+        Jacobian.DeleteValsRowi(total_index);
+      }
+    }
+  }
+  END_SU2_OMP_FOR
+}
+
+void CSpeciesSolver::BC_Outlet(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                               CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {
+  /// NOTE TK:: This is a strong impl whereas TurbSA and inceuler implement a weak version. Testing required.
+  /*--- Loop over all the vertices on this boundary marker ---*/
+
+  SU2_OMP_FOR_STAT(OMP_MIN_SIZE)
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    /* strong zero flux Neumann boundary condition at the outlet */
+    const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+
+    if (geometry->nodes->GetDomain(iPoint)) {
+      /*--- Allocate the value at the outlet ---*/
+      auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+      for (auto iVar = 0u; iVar < nVar; iVar++) Solution[iVar] = nodes->GetSolution(Point_Normal, iVar);
+      nodes->SetSolution_Old(iPoint, Solution);
+
+      LinSysRes.SetBlock_Zero(iPoint);
+
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        nodes->SetVal_ResTruncError_Zero(iPoint, iVar);
+      }
+
+      /*--- Includes 1 in the diagonal ---*/
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        auto total_index = iPoint * nVar + iVar;
+        Jacobian.DeleteValsRowi(total_index);
+      }
+    }
+  }
+  END_SU2_OMP_FOR
+}
+
+void CSpeciesSolver::BC_HeatFlux_Wall(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                                      CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {}
+
+void CSpeciesSolver::BC_Isothermal_Wall(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                                        CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {
+  BC_HeatFlux_Wall(geometry, solver_container, conv_numerics, visc_numerics, config, val_marker);
+}
```

Comment:
What you have for inlets is what you need here.

Reply:
I have to admit, I dont understand what you mean. Do you mean i need to handle walls explicitly for the species solver? A zero gradient for species is assumed on the wall and by doing nothing I think I fulfil that (In the FVM derivation for the boundary node the convective part is zero because v=0 and grad(Y)=0 makes the diffusion part go home). 

Or what do  you mean here 😬

Reply:
I was just thinking that in theory a species can also be a temperature, and that you may (for whatever reason) have a flux of this specie at walls or a Dirichlet (Isothermal) boundary condition for it.

But if you want species that can only be convected, then simply remove these empty implementations.

Reply:
yes, good point ... i did not think of that. There surely are some applications where these kinds of boundary conditions make sense. But as far as I can think this will not be the case for our applications @bigfooted ?

It would require new config options as well (which I would rather not do in this PR). Maybe write a comment in there with a query to Heatsolver and maybe CIncNSsolver(energy equation) how to tackle that ... that gives someone who wants to implement that already a good pointer

Reply:
In our applications we only have zero flux boundary conditions on the wall for the species. There are applications where this is different (charged species, semi-permeable walls, surface reactions) but we currently do not consider this. As this PR is large enough as it is, we can leave that implementation for a time when we actually need it.

Reply:
https://github.com/su2code/SU2/pull/1388/commits/318098926b3960ef4c442d4d4f741eb49faab79d I now removed the superfluous BC heatflux and isothermal functions bodies from CSpeciesSolver and added a little comment in the hpp that gives some future person maybe a pointer what to do and why that is